sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -207,7 +211,7 @@ case class TakeOrderedAndProject(
   private val ord: InterpretedOrdering = new InterpretedOrdering(sortOrder, child.output)
 
   // TODO: remove @transient after figure out how to clean closure at InsertIntoHiveTable.
-  @transient private val projection = projectList.map(new InterpretedProjection(_, child.output))
+  @transient private val projection = projectList.map(UnsafeProjection.create(_, child.output))
```

Comment:
Sorry, I still have a dumb question. When calling the `eval` of each of the specified expressions, how can we know they can process unsafe rows? Why the planner insert unsafe->safe conversion in the original design?